kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```
@@ -0,0 +1,700 @@
+				if lbingress.IP == addrs.String() {
+					continue
+				}
+			}
+
+			ipamArgs := ipam.AssignIPArgs{
+				IP:       addrs,
+				Hostname: api.VirtualLoadBalancer,
+				HandleID: &handle,
+				Attrs:    metadataAttrs,
+			}
+
+			err = c.calicoClient.IPAM().AssignIP(c.ctx, ipamArgs)
+			if err != nil {
+				log.WithField("ip", addrs).WithError(err).Warn("failed to assign ip to node")
+				return err
+			}
+
+			ingress = append(ingress, v1.LoadBalancerIngress{IP: addrs.String()})
+		}
+
+		svc.Status.LoadBalancer.Ingress = ingress
+		_, err = c.clientSet.CoreV1().Services(svc.Namespace).UpdateStatus(c.ctx, svc, metav1.UpdateOptions{})
+		return err
+	}
+
+	// Build AssignArgs based on Service IP family attr
+	num4 := 0
+	num6 := 0
+
+	for _, ipFamily := range svc.Spec.IPFamilies {
+		if ipFamily == v1.IPv4Protocol {
+			num4++
+		}
+		if ipFamily == v1.IPv6Protocol {
+			num6++
+		}
+	}
+
+	// Check if IP from ipfamily is already assigned, skip it as we're trying to assign only the missing one.
+	// This can happen when error happened during the initial assignment, and now we're trying to assign ip again from the syncIPAM func
+	ingressList := svc.Status.LoadBalancer.Ingress
+	if ingressList != nil {
+		for _, ingress := range ingressList {
+			if ip := cnet.ParseIP(ingress.IP); ip != nil {
+				if ip.To4() != nil {
+					num4 = 0
+				}
+				if ip.To16() != nil {
+					num6 = 0
+				}
+			}
+		}
+
+		if num4 == 0 && num6 == 0 {
+			return errors.New("No new IPs to assign, Service already has ipv4 and ipv6 address")
+		}
+	}
+
+	args := ipam.AutoAssignArgs{
+		Num4:        num4,
+		Num6:        num6,
+		IntendedUse: api.IPPoolAllowedUseLoadBalancer,
+		Hostname:    api.VirtualLoadBalancer,
+		HandleID:    &handle,
+		Attrs:       metadataAttrs,
+	}
+
+	if svc.Annotations[annotationIpv4Pools] != "" {
+		args.IPv4Pools, _ = c.resolvePools(parseAnnotation(svc.Annotations[annotationIpv4Pools]), true)
```

We shouldn't be ignoring errors anywhere